PeerioTechnologies/peerio-mobile
# Refactor chat list

#### Relevant info and issue/PR links 
 
https://app.clubhouse.io/peerio/story/15904/mobile-unread-message-bar-doesn-t-show-in-chat-list
https://app.clubhouse.io/peerio/story/15906/mobile-unread-message-bar-doesn-t-scroll

#### Testing instructions  

Login into account.
Receive messages in rooms or DMs which are invisible (offscreen). 
Check that unread indicators appear
Check that unread indicators appear after quickly switching between different routes (e.g. files-chats, files-chats) several times
Unread indicators must scroll precisely to the last or first unread item position, respectively.
Last unread item should be positioned at the bottom of the list, fully visible, after scroll.
First unread item should be positioned at the top of the list, fully visible, after scroll.
Unread indicators should appear when 80% of the unread item is invisible (cut off by render window).
Unread indicators should disappear when more than 20% of unread item is visible.
If there's a top drawer visible, scrolling should take into account the top drawer.

Possible regressions:
- chat list item height
- invite item height
- unread indicator animations
- top drawer positioning
- after-load list item rendering
- scrolling when top drawer is visible

Additional testing:

If you enable MockChatList in mocks/index.js, it is possible to easily test the scrolling behaviour on a big (200) list of items. It also enables guaranteed testing of scrolling to unrendered items. Generally, to scroll to unrendered items, you need at least 100 items in the list.

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
